### PR TITLE
Use graphql to fetch analysis Frameworks List

### PR DESCRIPTION
**Pull request by RobertKoikai**
- Addresses Use graphql to fetch list of analysis frameworks in projectEdit page #2727 
- Depends on :  (https://github.com/the-deep/server/pull/1369 )

## Changes

* Update the projectEdit  Framework page to fetch frameworks using graphql, 
* Remove all  REST API dependencies 
* Fix type related issues

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/views/ProjectEdit/Framework/index.tsx [hunk 1]
```diff
@@ -214,8 +229,11 @@ function ProjectFramework(props: Props) {
         offset,
     ]);
 
-    const totalFrameworksCount = frameworksResponse?.count ?? 0;
+    const totalFrameworksCount = projectUsersResponse?.analysisFrameworks?.totalCount ?? 0;
+    const AfList: FrameworkMini[] = projectUsersResponse?.analysisFrameworks?.results || [];
 
+    console.log('type one', typeof frameworkList);
+    console.log('type two >> ', projectUsersResponse?.analysisFrameworks?.results);
```
Comment (subinasr) [inline app/views/ProjectEdit/Framework/index.tsx]: Let's remove this.

File: app/views/ProjectEdit/Framework/index.tsx [hunk 2]
```diff
@@ -22,30 +22,47 @@ import {
...
+        $isCurrentUserMember: Boolean
+        $search: String,
+    ) {
+        analysisFrameworks(search: $search , isCurrentUserMember: $isCurrentUserMember) {
+         results {
+                    title
+                    id
+                    createdAt
+                }
+                totalCount
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: fix indentation

File: app/views/ProjectEdit/Framework/index.tsx [hunk 3]
```diff
@@ -22,30 +22,47 @@ import {
...
 interface FrameworkMini {
-    id: number;
+    id: string;
     title: string;
     createdAt: string;
 }
-
-const frameworkKeySelector = (d: FrameworkMini) => d.id;
-
+const frameworkKeySelector = (d: FrameworkMini) => parseInt(d.id, 10);
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: Why parseInt?

File: app/views/ProjectEdit/Framework/index.tsx [hunk 4]
```diff
@@ -156,37 +172,36 @@ function ProjectFramework(props: Props) {
...
+        let isCurrentUserMember;
+        if (delayedValue.relatedToMe === 'true') {
+            isCurrentUserMember = true;
+        } else if (delayedValue.relatedToMe === 'false') {
+            isCurrentUserMember = false;
+        } else {
+            isCurrentUserMember = undefined;
+        }
+        return {
+            isCurrentUserMember,
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: We should be able to directly pass relatedToMe as isCurrentUserMember. Can you check?

File: app/views/ProjectEdit/Framework/index.tsx [hunk 5]
```diff
@@ -156,37 +172,36 @@ function ProjectFramework(props: Props) {
...
+            search: delayedValue.search,
+            offset,
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: The order is missing.

File: app/views/ProjectEdit/Framework/index.tsx [hunk 6]
```diff
@@ -156,37 +172,36 @@ function ProjectFramework(props: Props) {
     } = useForm(schema, defaultFormValue);
 
     const delayedValue = useDebouncedValue(value);
-
     useEffect(() => {
         setOffset(0);
         setFrameworkList([]);
     }, [delayedValue]);
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: While you are here, this should also be removed. Can you make sure to update frameworkList and offset as soon as `value` is changed?

File: app/views/ProjectEdit/Framework/index.tsx [hunk 7]
```diff
@@ -214,8 +229,11 @@ function ProjectFramework(props: Props) {
         offset,
     ]);
 
-    const totalFrameworksCount = frameworksResponse?.count ?? 0;
+    const totalFrameworksCount = projectUsersResponse?.analysisFrameworks?.totalCount ?? 0;
+    const AfList: FrameworkMini[] = projectUsersResponse?.analysisFrameworks?.results || [];
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]:
```suggestion
    const frameworkList: FrameworkMini[] = projectUsersResponse?.analysisFrameworks?.results || [];
```
Ts variables should always be camelcase.

File: app/views/ProjectEdit/Framework/index.tsx (hunk 1 above)
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: Please make sure there are no console logs left before you sent the PR.
My recommendation is doing a self code review as soon as the PR is sent.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/ProjectEdit/Framework/index.tsx (hunk 1 above)
Comment (subinasr) [reply app/views/ProjectEdit/Framework/index.tsx]: For future, we have a checklist while creating PRs, checking those boxes only after confirming it is a recommended practice.

**Comment by RobertKoikai** [comment]
@AdityaKhatri ,

The PR has been updated.

Regarding the order please note that the backend for this is also modified to show frameworks that have been used to tag an entry within six months and list the filtered options with the most recently used framework on top.(See PR : https://github.com/the-deep/server/pull/1369).

cc ; @subinasr

File: app/views/ProjectEdit/Framework/index.tsx [hunk 8]
```diff
@@ -43,9 +46,26 @@ interface FrameworkMini {
...
 
+export const PROJECTS_AFS = gql`
+    query ProjectAnalysisFrameworks(
+        $isCurrentUserMember: Boolean
+        $search: String,
+    ) {
+        analysisFrameworks(
+            search: $search , 
+            isCurrentUserMember: $isCurrentUserMember
+            ) {
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: Fix indentation

File: app/views/ProjectEdit/Framework/index.tsx [hunk 9]
```diff
@@ -156,37 +175,22 @@ function ProjectFramework(props: Props) {
...
-        setFrameworkList([]);
-    }, [delayedValue]);
-
-    const queryForFrameworks = useMemo(() => ({
-        fields: ['id', 'title', 'created_at'],
-        offset,
-        order: 'created_at',
-        limit: 10,
+    const analysisFrameworkVariables = useMemo(() => ({
+        isCurrentUserMember: delayedValue.relatedToMe === 'true' ? true : undefined,
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]:
```suggestion
        isCurrentUserMember: delayedValue.relatedToMe,
```
Does this not work?

File: app/views/ProjectEdit/Framework/index.tsx [hunk 10]
```diff
@@ -214,8 +218,12 @@ function ProjectFramework(props: Props) {
...
     ]);
 
-    const totalFrameworksCount = frameworksResponse?.count ?? 0;
-
+    const totalFrameworksCount = projectAflist?.analysisFrameworks?.totalCount ?? 0;
+    const afList: FrameworkMini[] = (projectAflist?.analysisFrameworks?.results || [])
+        .map((item) => ({
+            ...item,
+            id: Number(item.id),
+        }));
```
Comment (AdityaKhatri) [inline app/views/ProjectEdit/Framework/index.tsx]: Lets not cast id to number.
After adopting graphql, all the id fields are string as well.
As this was in REST before, we used number to save selectedFramework, but it was eventually cast to string while checking whether its active or not.
Lets fix that there, and not cast.

Basically I recommend not casting it and fixing other places it causes issue to.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/ProjectEdit/Framework/index.tsx (hunk 10 above)
Comment (RobertKoikai) [reply app/views/ProjectEdit/Framework/index.tsx]: @AdityaKhatri ,
By  "fixing other places it causes issue to" does this mean fix the type issues in ListView in **deep-ui** ?
.

File: app/views/ProjectEdit/Framework/index.tsx (hunk 9 above)
Comment (RobertKoikai) [reply app/views/ProjectEdit/Framework/index.tsx]: @AdityaKhatri ,

To return ALL the analytical frameworks, relatedToMe should be neither true nor false. 

True returns **only** the AFs belonging to me .
False returns **only** the AFs that do not belong to me.

File: app/views/ProjectEdit/Framework/index.tsx (hunk 10 above)
Comment (AdityaKhatri) [reply app/views/ProjectEdit/Framework/index.tsx]: No, type issues will be in the rendererParams of ListView, so we need to fix it there. Let me know if a call is necessary.

File: app/views/ProjectEdit/Framework/index.tsx (hunk 9 above)
Comment (AdityaKhatri) [reply app/views/ProjectEdit/Framework/index.tsx]: Yes, and by default  relatedToMe is false or undefined? If its undefined then there shouldn't be any issue right?

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**